src/main.cpp

```diff
@@ -5789,6 +5841,10 @@ bool InitBlockIndex() {
     fTxIndex = GetBoolArg("-txindex", false);
     pblocktree->WriteFlag("txindex", fTxIndex);
 
+    // Use the provided setting for -maturityheightindex in the new database
+    fMaturityHeightIndex = GetBoolArg("-maturityheightindex", false);
+    pblocktree->WriteFlag("maturityheightindex", fMaturityHeightIndex);
```

Comment:
It is correctly written, but a ReadFlag() is missing at the node statup (see the analogous pblocktree->ReadFlag("txindex", fTxIndex);)